python/tank_vendor/shotgun_authentication/defaults_manager.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+
+
+class DefaultsManager(object):
+    """
+    This class allows the ShotgunAuthenticator to get some default values when
+    authenticating a user.
+    """
+
+    def is_host_fixed(self):
+        """
+        When doing an interactive login, this indicates if the user can choose
+        the host to connect to.
+
+        :returns: True is the host can't be edited, False otherwise,
+        """
+        return False
+
+    def get_host(self):
+        """
+        Override to provide a default host.
+
+        :returns: Default implementation returns None.
```

Comment:
For people subclassing, it will be useful to know what they are expected to return here. "Default implementation returns None." is already clear from the line below, something like `:returns: String containing a full shotgun url (e.g. 'https://foo.shotgunstudio.com') or None if this is not defined. If is_host_fixed returns True, this method cannot return None`